caas/kubernetes/provider/exec/exec_test.go

```diff
@@ -496,7 +496,40 @@ func (s *execSuite) TestModelNameToNameSpace(c *gc.C) {
 			}}, nil),
 	)
 
-	nsName, err := exec.ModelNameToNameSpace("controller", false, s.mockNamespaces)
+	nsName, err := exec.ModelNameToNameSpace("controller", "", false, s.mockNamespaces)
 	c.Assert(err, jc.ErrorIsNil)
 	c.Assert(nsName, gc.DeepEquals, "controller-k1")
 }
+
+func (s *execSuite) TestGetNamespaceForControllerModel(c *gc.C) {
+	ctrl := s.setupExecClient(c)
+	defer ctrl.Finish()
+
+	gomock.InOrder(
+		s.mockNamespaces.EXPECT().List(gomock.Any(), metav1.ListOptions{LabelSelector: "model.juju.is/name=controller"}).
+			Return(&core.NamespaceList{Items: []core.Namespace{
```

Comment:
could use a comment here saying what we're testing (i.e. multiple controllers running on the cluster)